sdks/java/core/src/main/java/org/apache/beam/sdk/testing/TestPipeline.java

```diff
@@ -292,8 +298,8 @@ public void evaluate() throws Throwable {
   public PipelineResult run() {
     checkState(
         enforcement.isPresent(),
-        "Attempted to run a pipeline while it's enforcement level was not set. Are you "
-            + "using TestPipeline without a @Rule annotation?");
+        "Is your TestPipeline declaration missing a @Rule annotation? Usage: "
```

Comment:
May I suggest the following course of actions:
1. We move forward with this PR so as to make things clearer and address the pain until we make the separation.
2. I'll file a JIRA to start working on separating the `TestPipeline` into 2 classes: 
    * One for the SDK and runner developers (with strict enforcement and the likes)
    * One for Beam users (a more lenient version of the former)